### PR TITLE
add autoscaling architecture docs

### PR DESCRIPTION
**Pull request by rodesai**

### CODE REVIEW

File: docs/concepts/autoscaling.md [hunk 1]
```diff
@@ -3,3 +3,19 @@ sidebar_position: 4
...
+
+## Overview
+
+Responsive automatically manages your applications to meet changing workload demands. You simply define a policy that specifies health goals for the application. The Responsive Controller combines the policy with metrics and status updates from the application and Responsive Operator to decide how to keep your cluster healthy. Then, it coordinates with the application and Responsive Operator to automatically execute remediating actions.
+
+## Architecture
+
+![Controller Architecture](/img/control-architecture.png "Control Architecture")
+
+You define a policy in a k8s custom resource (link to ref). The scaling policy defines a set of Diagnosers. A Diagnoser describes some health goals (e.g. keep latency low, keep utilization high) and constraints (dont scale past 10 replicas). Different diagnosers have different goals. For example, one diagnoser may have a goal of minimizing latency, while another diagnoser may have a goal of keeping utilization over some threshold. In conjunction, they define how to keep the cluster sized "just right" - large enough to keep up with your workload but keeping costs contained.
```
Comment (agavra) [inline docs/concepts/autoscaling.md:15]: > (link to ref)

we should just have a link to an empty page, that page can have "work in progress" on it

File: docs/concepts/autoscaling.md [hunk 2]
```diff
@@ -3,3 +3,19 @@ sidebar_position: 4
 ---
 
 # Autoscaling
+
+## Overview
+
+Responsive automatically manages your applications to meet changing workload demands. You simply define a policy that specifies health goals for the application. The Responsive Controller combines the policy with metrics and status updates from the application and Responsive Operator to decide how to keep your cluster healthy. Then, it coordinates with the application and Responsive Operator to automatically execute remediating actions.
```
Comment (agavra) [inline docs/concepts/autoscaling.md:9]: super nit, but can we newline at ~80 chars?

File: docs/concepts/autoscaling.md [hunk 3]
```diff
@@ -3,3 +3,19 @@ sidebar_position: 4
...
+
+## Architecture
+
+![Controller Architecture](/img/control-architecture.png "Control Architecture")
+
+You define a policy in a k8s custom resource (link to ref). The scaling policy defines a set of Diagnosers. A Diagnoser describes some health goals (e.g. keep latency low, keep utilization high) and constraints (dont scale past 10 replicas). Different diagnosers have different goals. For example, one diagnoser may have a goal of minimizing latency, while another diagnoser may have a goal of keeping utilization over some threshold. In conjunction, they define how to keep the cluster sized "just right" - large enough to keep up with your workload but keeping costs contained.
+
+Each instance of the policy specifies how to manage a single application. The policy spec references the Deployment or Statefulset resource that defines your application.
+
+You also install and run the Responsive Operator in your k8s cluster. It monitors the installed policies and the applications they point to, and updates the Responsive Controller with the current policy spec and status of the applications.
```
Comment (agavra) [inline docs/concepts/autoscaling.md:19]: we can link out to the operator doc in getting-started

File: docs/concepts/autoscaling.md (hunk 1 above)
Comment (agavra) [inline docs/concepts/autoscaling.md:15]: we might want to plug a bit of the "value" here (e.g. reduced operations, optimal cost/less hardware overpovisioning, never miss handling a burst of incoming events)

**Review by agavra — APPROVED** [review]
Thanks! Nice diagram

File: docs/concepts/autoscaling.md (hunk 2 above)
Comment (rodesai) [reply docs/concepts/autoscaling.md:9]: yep. easy enough with vim

File: docs/concepts/autoscaling.md (hunk 1 above)
Comment (rodesai) [reply docs/concepts/autoscaling.md:15]: yeah, forgot to do that

**Pull request merged**